### DISCUSSION TITLE
Removal of deprecated items in Tycho

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-tycho/tycho#1150] [General]
I'd like to suggest that with 3.x on we adjust our retention policy to remove deprecated things from mojos on the ext **minor** version release so we can do:

1. deprecate (and note in the release notes) in release X
2. remove (and note in the release notes) in release X+1

I think, given that we offer snapshots for preview next release and people can always choose to use an older release, it is fine to speed up removal of deprecated features.

**Comment by akurtakov** [comment]
What would the definition be of major be in this case? Why not doing major releases when something is removed? I would prefer having more frequent major releases but keeping it clearer that smth broke.
If the concern is possible delay due to time 3.0 takes to shape - it's due to the fact that breaking changes hasn't happened for years.

**Reply by laeubi** [reply]
> What would the definition be of major be in this case?

For major releases e.g.:

- require a new java version to run the build
- removal of mojos
- require never maven versions

> I would prefer having more frequent major releases

I really don't want to end with Tycho 23 :-)

> If the concern is possible delay due to time 3.0 takes to shape - it's due to the fact that breaking changes hasn't happened for years.

I more think about e.g. we found that an option of a mojo is no more useful, we can get rid of it.
